### PR TITLE
Update migraphx for rocm 3.8

### PR DESCRIPTION
**Pull request by srekolam**
Updated the checksum for rocm-3.7.0 and rocm-3.8.0 based on PR https://github.com/spack/spack/pull/18719

Closes #18719

### CODE REVIEW

**Comment by srekolam** [comment]
I did the test for spack install migraphx@3.5.0 and spack install migraphx@3.7.0 on centos-8.x , gcc8.3.1 . For spack install migraphx@3.8.0 , i have to test it .

File: var/spack/repos/builtin/packages/migraphx/package.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    maintainers = ['srekolam', 'arjun-raj-kuppala']
+
+    def url_for_version(self, version):
+        # Fix up a typo in the 3.5.0 release.
+        if version == Version('3.5.0'):
+            return "https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/archive/rocm-3.5.0.tar.gz"
+        if version == Version('3.7.0'):
+            return "https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/archive/0.7.tar.gz"
+        if version == Version('3.8.0'):
+            return "https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/archive/0.8.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/migraphx/package.py]:
```suggestion
        url = "https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/archive/{0}.tar.gz"
        if version > Version('3.5'):
            return url.format(version)
        else:
            return url.format('rocm-' + version)
```

Comment (srekolam) [reply var/spack/repos/builtin/packages/migraphx/package.py]: done. i have updated the file with the requested changes

**Comment by srekolam** [comment]
@adamjstewart  , can you please review and approve.

File: var/spack/repos/builtin/packages/migraphx/package.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Migraphx(CMakePackage):
+    """ AMD's graph optimization engine."""
+
+    homepage = "https://github.com/ROCmSoftwarePlatform/AMDMIGraphX"
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/migraphx/package.py:14]: I would still include a URL up here, it's used by `spack versions` and `spack checksum` to locate new versions.

Comment (srekolam) [reply var/spack/repos/builtin/packages/migraphx/package.py:14]: Just wanted to know if you meant the below
 homepage = "https://github.com/ROCmSoftwarePlatform/AMDMIGraphX"
    url = "https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/archive/rocm-3.5.0.tar.gz"

    maintainers = ['srekolam', 'arjun-raj-kuppala']

    def url_for_version(self, version):
        url = "https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/archive/{0}.tar.gz"
        if version > Version('3.5'):
            return url.format(version)
        else:
            return url.format('rocm-' + version)

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/migraphx/package.py:14]: Yes, just put it below the homepage

File: var/spack/repos/builtin/packages/migraphx/package.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
+
+    depends_on('cmake@3:', type='build')
+    depends_on('protobuf', type='link')
+    depends_on('blaze', type='build')
+    depends_on('nlohmann-json', type='link')
+    depends_on('py-pybind11', type='build')
+    depends_on('msgpack-c', type='link')
+    depends_on('msgpack-c', type='build', when='@3.8.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/migraphx/package.py]: Can you explain this a bit? Spack has trouble merging deptypes, so you may need to be more explicit about this. Is it link for all versions and build/link for the latest version? Or is it link for older versions and build for newer?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**